framework/src/controller/bus.ts

```diff
@@ -415,21 +420,71 @@ export class Bus {
 		}
 	}
 
-	private async _setupIPCServer(): Promise<void> {
-		await this._ipcServer.start();
+	private async _setupIPCInternalServer(): Promise<void> {
+		await this._internalIPCServer?.start();
 
-		const listenToEvents = async () => {
-			for await (const [_event, eventData] of this._ipcServer.subSocket) {
+		const listenToEvents = async (subSocket: Subscriber) => {
+			for await (const [_event, eventData] of subSocket) {
 				this.publish(eventData.toString());
 			}
 		};
 
-		listenToEvents().catch(err => {
+		const listenToRPC = async (rpcServer: Router) => {
```

Comment:
Would it be better to define this and `listenToEvents` functions outside?

Reply:
extracted to separate functions